File: CoreJava/SOLID-app/SRP/com/techlab/violation/solution/Invoice.java
package com.techlab.violation.solution;

public class Invoice {

	private int id;
	private String name;
	private double amount;
	private float discountPercentage;
	private float GSTPercentage;

	public Invoice(int id, String name, double amount, float discountPercentage, float gSTPercentage) {
		this.id = id;
		this.name = name;
		this.amount = amount;
		this.discountPercentage = discountPercentage;
		GSTPercentage = gSTPercentage;
	}

	public int getId() {
		return id;
	}

	public String getName() {
		return name;
	}

	public double getAmount() {
		return amount;
	}

	public float getDiscountPercentage() {
		return discountPercentage;
	}

	public float getGSTPercentage() {
		return GSTPercentage;
	}

	public void calculateGSTAmount() {
		amount = amount + (amount * (GSTPercentage/100f));
	}

	public void calculateDiscountAmount() {
		amount = amount + (amount * ( discountPercentage/100f));
	}

	
}

File: C#Basics/Invoice-Constructor-app/Invoice.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Invoice_Constructor_app
{
    class Invoice
    {
		private int id;
		private string name;
		private double amount;
		private double discountPercentage;
		private readonly float GST = 5f;
		private double discountedAmount;

		//parameterized constructor
		public Invoice(int id, String name, double amount, double discountPercentage)
		{
			this.id = id;
			this.name = name;
			this.amount = amount;
			this.discountPercentage = discountPercentage;
		}

		//default constructor
		public Invoice()
		{
			id = 0;
			name = "notDefined";
			amount = 0.0;
			discountedAmount = 0;
		}

		public int getID()
		{
			return id;
		}

		public String getName()
		{
			return name;
		}

		public double getAmount()
		{
			return amount;
		}

		public double getDiscountPrecentage()
		{
			return discountPercentage;
		}

		public double calculateDiscount(double amount, double discountPercentage)
		{
			discountedAmount = (amount - (discountPercentage / 100 * amount));
			return (discountedAmount);
		}


		public float calculateGST()
		{
			return (float)(GST / 100 * discountedAmount);
		}

		public double calculateTotalCost(double Gst)
		{
			return (this.calculateDiscount(amount, discountPercentage) + Gst);
		}

	}
}